runtime/src/main/java/org/corfudb/runtime/CheckpointWriter.java

```diff
@@ -47,8 +47,15 @@
     private LocalDateTime startTime;
     private long startAddress;
     private long endAddress;
+    @Getter
     private long numEntries = 0;
     private long numBytes = 0;
+    // SMR entries total size falling on EXACTLY on the write size limit set will create problems
+    // Making sure there is 2% room, a safety guard to account for checkpoint entry overhead
+    // (metadata)
+    @Setter
+    @Getter
+    private double BATCH_THRESHOLD_PERCENTAGE = 0.98;
```

Comment:
This seems arbitrary. I would just suggest using `sizeOf` on an empty checkpoint to calculate it exactly.

Reply:
Resolving this after our discussion. For now, we agree to go with 0.95